.github/workflows/release-preview.yaml

```diff
@@ -0,0 +1,177 @@
+name: Release Preview
+
+on:
+  workflow_call:
+
+jobs:
+  preview:
+    name: Preview Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Setup branch for semantic-release
+        run: |
+          # Explicitly checkout to the PR branch by name
+          git checkout -B ${{ github.event.pull_request.head.ref }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
+
+      - name: Setup preview config
+        run: |
+          # Update preview config with the PR branch name
+          sed -i.bak "s|BRANCH_PLACEHOLDER|${{ github.event.pull_request.head.ref }}|g" .preview-releaserc.json
+          rm .preview-releaserc.json.bak
+
+      - name: Run semantic-release (dry-run)
+        id: semantic
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GIT_COMMITTER_NAME: "github-actions[bot]"
+          GIT_COMMITTER_EMAIL: "github-actions[bot]@users.noreply.github.com"
+          GIT_AUTHOR_NAME: "github-actions[bot]"
+          GIT_AUTHOR_EMAIL: "github-actions[bot]@users.noreply.github.com"
+        run: |
+          # Unset GitHub Actions environment variables that interfere with semantic-release
+          unset GITHUB_REF
+          unset GITHUB_REF_NAME
+          unset GITHUB_HEAD_REF
+          unset GITHUB_BASE_REF
+
+          # Set them to what we want
+          export GITHUB_REF="refs/heads/${{ github.event.pull_request.head.ref }}"
+          export GITHUB_REF_NAME="${{ github.event.pull_request.head.ref }}"
+
+          # Temporarily use preview config
+          mv .releaserc.json .releaserc.json.main
```

Comment:
why are we doing this? can't we explicitly mention which file to be used in semantic-release?

Reply:
I don't release-preview being used, is that to be removed?